### ISSUE TITLE
mpv shaders : nlmeans will heavily smear textures

### ISSUE THREAD

**Issue by dyphire** [AN3223/dotfiles#11]
When using the shader of nlmeans to test its noise reduction effect in some videos, I noticed that it produces severe texture smearing behavior, which did not occur in its filter version.
test file: [sample.zip](https://github.com/AN3223/dotfiles/files/12206808/sample.zip)

nlmeans.glsl
![image](https://github.com/AN3223/dotfiles/assets/61936050/3b7d3eb9-4f5a-4215-9d1a-5fea51260126)

vf_nlmeans
![image](https://github.com/AN3223/dotfiles/assets/61936050/718b460b-0824-47c6-9820-401a2d5b4310)

**Comment by AN3223** [comment]
I agree, this is too much blur for this scene. I just pushed a commit (bda61d72b3dc898ba13d07038ebff289c4c74051) that adds a light-noise tuned profile [here](https://github.com/AN3223/dotfiles/blob/master/.config/mpv/shaders/dev/experimental/nlmeans_light.glsl), let me know what you think. I'll also look into building an slightly lighter profile.

`nlmeans_light.glsl` output:

![mpv-shot0001](https://github.com/AN3223/dotfiles/assets/35916076/c17ba806-902a-4431-93fe-d0de7d2d23ee)

**Comment by dyphire** [comment]
> I agree, this is too much blur for this scene. I just pushed a commit ([bda61d7](https://github.com/AN3223/dotfiles/commit/bda61d72b3dc898ba13d07038ebff289c4c74051)) that adds a light-noise tuned profile [here](https://github.com/AN3223/dotfiles/blob/master/.config/mpv/shaders/dev/experimental/nlmeans_light.glsl), let me know what you think. I'll also look into building an slightly lighter profile.
> 
> `nlmeans_light.glsl` output:

The effect of `nlmeans_light.glsl` has improved in my test, although significant texture loss can still be seen

**Comment by AN3223** [comment]
I pushed a commit that fixes a bug which was creating extra blur. Here is the new state of `nlmeans_light.glsl`:

![mpv-shot0001](https://github.com/AN3223/dotfiles/assets/35916076/63f2a3e9-2b24-48a1-9a90-fd3ec4866e68)

Focusing only on the blinds in the background, it now appears that the difference between `-vf=nlmeans` and `nlmeans_light.glsl` is quite small.

There is low frequency noise on the left head and the pink suit that is handled well by `-vf=nlmeans` but not handled well by any of my nlmeans shaders. So I built an nlmeans profile to replicate the effect of `-vf=nlmeans` (it can be found [here](https://github.com/AN3223/dotfiles/blob/master/.config/mpv/shaders/HQX/nlmeans_ffmpeg_eqv.glsl)) and the result is virtually identical to `-vf=nlmeans` (I measured an SSIM of `0.997064` where `1.0` would mean identical):

![mpv-shot0001](https://github.com/AN3223/dotfiles/assets/35916076/a90f229e-deb7-4fd6-863f-f093d2ed416e)

So everything appears to be working correctly now. I'll eventually make light variants for LQ/HQ/sharpen/sharpen+denoise/etc., and the new `ffmpeg_eqv` profile will always be there too.
